torch/_dynamo/backends/distributed.py

```diff
@@ -321,14 +336,20 @@ def run_node(self, n: Node) -> Any:
 
             # Finally, we have to produce inputs for use compiling the next submodule,
             # and these need to be FakeTensors, so we execute the module under fake_mode
-            with self.fake_mode:
-                return curr_submod(*new_args, **kwargs)
+            with self.fake_mode, mock.patch.object(
+                self.fake_mode, "allow_non_fake_inputs", True
+            ):
+                if has_tracing_context:
+                    return compiled_submod_real(*new_args, **kwargs)
```

Comment:
This line is in a stack trace from the moco model in torchbench ( https://www.internalfb.com/phabricator/paste/view/P1189842447 ) . Looks like the fake tensors passed to the compiled submodule are then being passed to a deferred call of cudagraphify, which seems to expect real tensors.

Reply:
passes now